### ISSUE TITLE
VoxelGeneratorScript does not work

### ISSUE THREAD

**Issue by Anyeos** [Zylann/godot_voxel#626]
**Describe the bug**
VoxelGeneratorScript does not work at all. I tried a lot of things, from using a noise, to only calling set_voxel to calling set_voxel_f and so. There are no way to make it work. It does not take in account at all the data. I can set 1, 0, 100, 10000, 10.0, etc but it does not do anything.

**To Reproduce**
Use VoxelGeneratorScript.

Example:
```
func _generate_block(out_buffer, origin_in_voxels, lod):
	out_buffer.fill_f(10.0, VoxelBuffer.CHANNEL_SDF)

func _get_used_channels_mask() -> int:
	return 1 << VoxelBuffer.CHANNEL_SDF
```

**Expected behavior**
Show the voxels.

**Environment**
- OS: Ubuntu 20.04
- Vulkan API 1.3.242 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 3070 Ti
- Godot 4.2 stable
- Module version an old one and the last from github (commit 19faf4b6580ef469cf74b1047870508eec2c1aa1)
- Forward+

**Comment by Zylann** [comment]
```gdscript
out_buffer.fill_f(10.0, VoxelBuffer.CHANNEL_SDF)
```

Assuming you are using the Transvoxel mesher, this line fills all space with signed distance 10, which is outside matter, so you will see nothing. In fact, if you set the same value everywhere, you will rarely every see anything, no matter the value. Because it either means all space is air when > 0, or all space is matter when < 0, so no surface will ever show up. For surfaces to show up, voxels have to vary across 0.

As shown in https://voxel-tools.readthedocs.io/en/latest/smooth_terrain/#signed-distance-fields, to write a smooth voxels generator, you need an understanding of signed distance fields.

For example, a generator producing a sphere of radius 10 centered at the origin, would look like this:
```gdscript
func _generate_block(out_buffer, origin_in_voxels, lod):
	for rz in out_buffer.get_size().z:
		for rx in out_buffer.get_size().x:
			for ry in out_buffer.get_size().y:
				var pos_world := Vector3(origin_in_voxels) + Vector3(rx, ry, rz)
				var sd := pos_world.distance_to(Vector3(0,0,0)) - 10.0
				out_buffer.set_voxel_f(sd, VoxelBuffer.CHANNEL_SDF)
```

Or a plane at altitude 5:
```gdscript
func _generate_block(out_buffer, origin_in_voxels, lod):
	for rz in out_buffer.get_size().z:
		for rx in out_buffer.get_size().x:
			for ry in out_buffer.get_size().y:
				var pos_world := Vector3(origin_in_voxels) + Vector3(rx, ry, rz)
				var sd := pos_world.y - 5.0
				out_buffer.set_voxel_f(sd, VoxelBuffer.CHANNEL_SDF)
```

However if you intented to use blocky voxels, your code would look a bit different.

**Comment by Anyeos** [comment]
Yes, sorry, I noted it recently. I was following the C code under the other generators and I missunderstood this. But anyway maybe there can be some notation or doc that explains how is it. Thank you for your explanation that makes me understand it better.
